comictaggerlib/ui/customwidgets.py

```diff
@@ -112,3 +122,256 @@ def toggleItem(self, index: int) -> None:
+        y = option.rect.y()
+        h = option.rect.height()
+        checkBoxTopLeftCorner = QPoint(2, int(y + h / 2 - checkBoxRect.height() / 2))
+
+        return QRect(checkBoxTopLeftCorner, checkBoxRect.size())
+
+    def itemClicked(self, index: QModelIndex, pos: QPoint) -> None:
+        item_rect = self.combobox.view().visualRect(index)
+        checked = index.data(Qt.CheckStateRole)
+        button_up_rect = self._button_up_rect(item_rect)
+        button_down_rect = self._button_down_rect(item_rect)
+
+        if checked and button_up_rect.contains(pos):
+            self.buttonClicked.emit(index, ClickedButtonEnum.up)
+        elif checked and button_down_rect.contains(pos):
+            self.buttonClicked.emit(index, ClickedButtonEnum.down)
+        else:
+            self.buttonClicked.emit(index, ClickedButtonEnum.main)
+
+    def sizeHint(self, option: QtWidgets.QStyleOptionViewItem, index: QModelIndex) -> QSize:
+        return QSize(10, 35)
+
+
+# Multiselect combobox from: https://gis.stackexchange.com/a/351152 (with custom changes)
+class CheckableOrderComboBox(QtWidgets.QComboBox):
+    itemClicked = pyqtSignal(QModelIndex, QPoint)
+    dropdownClosed = pyqtSignal(list)
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.setItemDelegate(ReadStyleItemDelegate(self))
+        # Prevent popup from closing when clicking on an item
+        self.view().viewport().installEventFilter(self)
+
+        # Go on a bit of a merry-go-round with the signals to avoid custom model/view
+        self.itemDelegate().buttonClicked.connect(self.buttonClicked)
+
+        # Keeps track of when the combobox list is shown
+        self.justShown = False
+
+    def buttonClicked(self, index: QModelIndex, button: ClickedButtonEnum) -> None:
+        if button == ClickedButtonEnum.up:
+            self.moveItem(index.row(), up=True)
+        elif button == ClickedButtonEnum.down:
+            self.moveItem(index.row(), up=False)
+        else:
+            self.toggleItem(index.row())
+
+    def resizeEvent(self, event: Any) -> None:
+        # Recompute text to elide as needed
+        super().resizeEvent(event)
+        self._updateText()
+
+    def eventFilter(self, obj: Any, event: Any) -> bool:
+        # Allow events before the combobox list is shown
+        if obj == self.view().viewport():
+            # We record that the combobox list has been shown
+            if event.type() == QEvent.Show:
+                self.justShown = True
+            # We record that the combobox list has hidden,
+            # this will happen if the user does not make a selection
+            # but clicks outside of the combobox list or presses escape
+            if event.type() == QEvent.Hide:
+                self._updateText()
+                self.justShown = False
+                self.dropdownClosed.emit(self.currentData())
```

Comment:
I thought this made sense. Doing the same for modify would as well. Could also change modify to dropdown close too instead of item check change. Although I do like seeing the fields open and close when clicking the modify styles on and off.